test/quantization/fx/test_equalize_fx.py

```diff
@@ -135,220 +210,296 @@ def test_input_weight_eq_observer(self, input_qdtype, input_qscheme, weight_qdty
-            def __init__(self):
-                super().__init__()
-                self.linear1 = Linear()
-                self.linear2 = Linear()
-
-            def forward(self, x):
-                x = self.linear1(x)
-                x = self.linear2(x)
-                return x
-
-        tests = [(LinearModule, default_equalization_qconfig_dict),
-                 (Linear2FP32Module, default_equalization_qconfig_dict),
-                 (FunctionalLinearModule, default_equalization_qconfig_dict),
-                 (FunctionalLinear2FP32Module, default_equalization_qconfig_dict),
-                 (Linear2Module, default_equalization_qconfig_dict),
-                 (FunctionalLinear2Module, default_equalization_qconfig_dict)]
-
-        for (M, equalization_qconfig_dict) in tests:
+        tests = [SingleLayerLinearModel, Linear2FP32Module, TwoLayerLinearModel,
+                 FunctionalLinearModule, FunctionalLinear2FP32Module, FunctionalLinear2Module]
+
+        x = torch.rand((5, 5))
+        for M in tests:
             m = M().eval()
-            x = torch.tensor([[1.0, 2.0], [2.0, 2.5], [4.5, 6.0]])
-            prepared = prepare_fx(m, qconfig_dict, equalization_qconfig_dict=equalization_qconfig_dict)
+            prepared = prepare_fx(copy.deepcopy(m), qconfig_dict, equalization_qconfig_dict=default_equalization_qconfig_dict)
             output = prepared(x)
 
             convert_ref = _convert_equalization_ref(prepared)
             convert_ref_output = convert_ref(x)
 
-            m = M().eval()
-            prepared = prepare_fx(m, qconfig_dict, equalization_qconfig_dict=equalization_qconfig_dict)
+            prepared = prepare_fx(m, qconfig_dict, equalization_qconfig_dict=default_equalization_qconfig_dict)
             prepared(x)
-            convert_fx(prepared)  # Check if compile?
+            convert_fx(prepared)  # Check if compile
             self.assertEqual(output, convert_ref_output)
+
+    def calculate_equalization_scale_ref(self, x, w):
+        """ Calculates the equalization scale based on the input and weight
+        """
+        min_inputs = x.min(axis=0)
+        max_inputs = x.max(axis=0)
+
+        min_weights_col = w.min(axis=0)
+        max_weights_col = w.max(axis=0)
+
+        equalization_scale = np.sqrt((max_weights_col - min_weights_col) /
+                                     (max_inputs - min_inputs))
+        return equalization_scale
+
+    def get_expected_eq_scales(self, model, x):
+        """ For each module in the graph, we want to calculate the equalization
+        scale at that point
+        """
+        exp_eq_scales = []
+        for _, module in model.named_children():
+            weight = module.weight.detach().numpy()
+            bias = module.bias.detach().numpy()
+
+            eq_scale = self.calculate_equalization_scale_ref(x, weight)
+            exp_eq_scales.append(eq_scale)
+
+            x = x @ weight.T + bias
```

Comment:
This would work only for single/connected linear modules, right? Should we make that clearer?

Reply:
Yes, I can leave a comment on the functions.